lib/execution_engine2/utils/KafkaUtils.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+"""Module to provide kafka handlers for internal logging facility."""
+
+import json
+import logging
+
+from execution_engine2.db.models.models import Status
+
+logging.basicConfig(level=logging.INFO)
+from dataclasses import dataclass
+
+from confluent_kafka import Producer
+
+KAFKA_EVENT_TYPE = "job_status_update"
+CONDOR_EVENT_TYPE = "condor_request"
+DEFAULT_TOPIC = "ee2"
+
+
+def send_kafka_update_status(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdate)
+
+
+def send_kafka_condor_update(data):
+    send_message_to_kafka(data=data, data_class=KafkaCondorCommandUpdate)
+
+
+def send_kafka_update_start(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdateStartJob)
+
+
+def send_kafka_update_finish(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdateFinishJob)
+
+
+def send_kafka_update_cancel(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdateCancelJob)
+
+
+@dataclass
+class KafkaStatusUpdate:
+    job_id: str = None
+    previous_status: str = None
+    new_status: str = None
+    scheduler_id: int = None
+    topic: str = DEFAULT_TOPIC
+    event_type: str = KAFKA_EVENT_TYPE
+    error: bool = False
+
+    def __post_init__(self):
+        if self.job_id is None:
```

Comment:
You don't need this manual type check if you take out the `= None` above in the field definition. So if you just have `job_id: str`, and you don't pass in a job_id when instantiating the object, then an error will get raised.

Reply:
@jayrbolton If I do that, I cannot inherit the dataclasses